### PR TITLE
matterbridge: Add initial config

### PR DESCRIPTION
**Pull request by RyanSquared**
This uses matterbridge to bridge a channel on Hashbang IRC with a channel on Freenode.

TODO eventually: just bridge directly to Matrix, I believe you can have multiple bridges in one setup. I'm not doing this initially to avoid duplicating messages with the already-existing bridge.

### CODE REVIEW

**Comment by RyanSquared** [comment]
would appreciate a :+1: from @daurnimator before merging

File: matterbridge/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+namespace: matterbridge
+resources:
+- namespace.yaml
+- resources.yaml
+generators:
+- secret-generator.yaml
+images:
+- name: 42wim/matterbridge
+  digest: sha256:22c5235c8ad2f28591b3f6d6357e0b6a42700a362faa671fed84718f0f3b70ca
+
```
Comment (daurnimator) [inline matterbridge/kustomization.yaml:12]: extra whitespace

File: matterbridge/secret-generator.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: viaduct.ai/v1
+kind: ksops
+metadata:
+  name: matterbridge-secrets
+files:
+- ./matterbridge-config.enc.yaml
```
Comment (daurnimator) [inline matterbridge/secret-generator.yaml:6]: usually wouldn't have the `./`

File: matterbridge/resources.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    metadata:
+      labels:
+        app: matterbridge
+    spec:
+      containers:
+      - name: matterbridge
+        image: 42wim/matterbridge
+        volumeMounts:
+        - name: matterbridge-config
+          mountPath: "/etc/matterbridge"
```
Comment (daurnimator) [inline matterbridge/resources.yaml:21]: can drop the quotes

File: matterbridge/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: matterbridge
+resources:
+- namespace.yaml
```
Comment (daurnimator) [inline matterbridge/kustomization.yaml:5]: no need to have the namespace

File: matterbridge/matterbridge-config.enc.yaml [hunk 5]
```diff
@@ -0,0 +1,68 @@
+apiVersion: v1
+kind: Secret
+stringData:
+    matterbridge.toml: ENC[AES256_GCM,data:bgMn/C9uGMQVBcpdN1qk5qcNdKuLgmKoK34ike5pO5GK8QRb2yIA7kDPtkmqMVE/4skKLwAyB/T4aKSFC3YlaV7p/WXlitr7jy43Gi7yCP5lEtXcvJFTXX6yFUdmhGXvNM8yaR2bs5Pgohz+iaBvP9Wm3vOM9YVrCZ0dwtQKMCv531gYfi0e9moL8kwbN342P+cKO6JpiytlU27kcrDmMU633myOajd9ZMW2PahyIcMqT60+fhc7gQm9bnBqcoteJ1b98/I8YaIWjl6VPS5QNHANJrcjPgloFr0dabBmYbPQT57vO7e1dREhB03MJY1AvZEgTavQ6qSHa3s3eFxXaijRjjbCIiqz7HLfmCtDeutLfCzivruhLtd+mOyH+v7TK7KTDbK/MuekY5QlnDBZQ+ga6zq3VUFYBc1DW0xnbbkmcqLRvywOLE7nhtuwJnyDUOGWVoZSvSXLnAvGiVMhxkCbPZvvfcWrOnCs9C8BNuVAiS9JEDTSjRUYVqilTTcoxlCLCvPXCf9lY7RHcrAm92vTaPb/F0Xf78DuXtrq+qKoWwLJizLgmd+3KEza3s8VWUk8xj1kAH9/w+XzIdhYI9euZIMKMABEeVyWJm9fjJNnbY1eAbU1oge56RDI9z/nQIT8955IE1ZpfH+1Bj1TcQTfs6NWbGpyYLNJFBXeMeLt6K03ytUFjWhB/xzgbJZqKsKV3trQx2ff8UdOrifyGczrLeC706ywIP0821BHIRRRj4rFyc+tTvULAa/N6W1jfPQNL/kQ4PSHN6Vjgs7NmDoBweNafJhWxy+N4bUvBWHmYjK2g1K455ySt78QUfb+gFjEkJTUejmATe91HeKyaneij12qQaH/79NwE0JLHY9g+iEZZaw9v5QQahw73/NC0KZBWQ5p97jA0Z1ba1QAe7ZuH9/h5/xKlcOb+ggOpronNZFxJ0+sUiMSOIz27kfPEvU59xX0mNV7IUwtsfyAu7kxhs3RwWrofs3h3jwYVZt8P/sJ6u0FKTGZ,iv:z7+f8tzarpL7KDFFLzNcb+pVHWdk8Xfm8QAypHIK1FM=,tag:tvoV97oQxzpEtIrc75WyNw==,type:str]
```
Comment (daurnimator) [inline matterbridge/matterbridge-config.enc.yaml:4]: is there any nice way to break this out?

**Review by daurnimator — CHANGES_REQUESTED** [review]

File: matterbridge/matterbridge-config.enc.yaml (hunk 5 above)
Comment (RyanSquared) [reply matterbridge/matterbridge-config.enc.yaml:4]: not without an envsubst init container; i figured this was simpler. if you'd like i can make one

Comment (RyanSquared) [reply matterbridge/matterbridge-config.enc.yaml:4]: correction: possible, but poorly documented. will test.

File: matterbridge/matterbridge-config.enc.yaml [hunk 6]
```diff
@@ -0,0 +1,72 @@
+apiVersion: v1
+kind: Secret
+metadata:
+    name: matterbridge-secret
```
Comment (daurnimator) [inline matterbridge/matterbridge-config.enc.yaml:4]: you'll want a needs-hash annotation on this.

**Review by daurnimator — CHANGES_REQUESTED** [review]

**Review by daurnimator — APPROVED** [review]

**Pull request closed without merging**